services/provider/server/server.go

```diff
@@ -1079,3 +1090,54 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) RequestMaintenanceMode(ctx context.Context, req *pb.RequestMaintenanceModeRequest) (*pb.RequestMaintenanceModeResponse, error) {
+	// Get storage consumer resource using UUID
+	consumerObj, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		klog.Error(err)
+		return nil, status.Errorf(codes.NotFound, "Failed to get storageConsumer resource")
+	}
+
+	if req.RequestFailover {
+		util.AddAnnotation(consumerObj, util.RequestFailoverMaintenanceModeAnnotation, "")
+	} else {
+		delete(consumerObj.GetAnnotations(), util.RequestFailoverMaintenanceModeAnnotation)
+	}
+
+	err = s.consumerManager.Update(ctx, consumerObj)
+	if err != nil {
+		klog.Error(err)
+		return nil, status.Errorf(codes.Internal, "Failed to update storageConsumer resource")
+	}
```

Comment:
Should we always update? even if there is no change?

Reply:
No, added logic to update when there is change